src/dune_rules/pkg_rules.ml

```diff
@@ -615,6 +689,8 @@ module Action_expander = struct
             in
             env, update :: updates)
       in
+      let expander = { expander with env } in
```

Comment:
Isn't this a separate bug? Did you end up writing a test for this?

Reply:
I didn't, because to show the expander is not updated with the bindings to execute the action it needs to access the expander bindings. I looked at it and it seems that withenv does the substitution first so by the time the action is executed it doesn't really matter what bindings the expander contains.

I can remove this drive-by fix, show it is not working in the cram test here and make a separate bug fix PR if that is preferable.

Reply:
Notably I wrote a test running nested `withenv`s, hoping the inner `withenv` wouldn't see the outer bindings but it works on current `main`.